src/main/java/org/study/periodicals/filters/AuthorizationFilter.java

```diff
@@ -9,28 +10,49 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class AuthorizationFilter extends HttpFilter {
 
     private UserAuthorizationService userAuthorizationService;
 
     private FilterConfig filterConfig;
 
+    private SecurityConfig securityConfig;
+
     @Override
     protected void doFilter(HttpServletRequest req, HttpServletResponse res, FilterChain chain) throws IOException, ServletException {
 
-
         HttpSession session = req.getSession(false);
 
         boolean userAuthorized = userAuthorizationService.isUserAuthorized(session.getId());
 
+        String contextPath = filterConfig.getServletContext().getContextPath();
 
         if (!userAuthorized) {
-            String contextPath = filterConfig.getServletContext().getContextPath();
             res.sendRedirect(contextPath + "/loginPage");
-            return;
-        }
 
+        } else {
+            String currentUri = req.getRequestURI();
+            securityConfig.getRolesToUrls().get(1).forEach(it -> {
+                boolean matches = it.pattern().matches(currentUri);
+                if (!matches){
+                    try {
+                        res.sendRedirect(contextPath + "/personal/*");
```

Comment:
res.sendRedirect(contextPath + "/some_kind_of_error_page/*").
where "some_kind_of_error_page" is page, which would show user that he/she are not authorized to view this page